YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs

```diff
@@ -71,23 +127,131 @@ public void UnshowPlayer()
+    /// <returns></returns>
+    private IEnumerator SetPlayerOutOfHouse()
+    {
+        Debug.Log ("Starting coroutine OutHouse");
+        aCoroutineRuns = true;
+
+        // If still active, deactivate the PressSpace Canvas
+        if (PressSpace != null)
+        {
+            PressSpace.gameObject.SetActive(false);
+        }
+
+        // get player out of house again
+        UnshowPlayer();
+        player.SetActive(true);
+        player.transform.position = playerAppearPosition;
+        isPlayerInside = false;
+        // Notify the endlevel controller that we left the home
+        endlevel.NotifyPlayerLeftHome();
+       
+        // If the player started inside, this is now over
+        // and we need to set playerStartsInside to false.
+        playerStartsInside = false;
+
+        aCoroutineRuns = false;
+
+        yield return 0;
+    }
+
     /// <summary>
     /// Puts the player in the house
     /// Healthy player or exposed player depending on condition
-    /// Notifies end level controller
+    /// Notifies end level controller if wished
     /// All with a little bit of time delay to make it look good
+    /// Calls SetPlayerOutOfHouse
     /// </summary>
-    private IEnumerator SetPlayerInHouse(GameObject p)
+    /// <returns></returns>
+    private IEnumerator SetPlayerInHouse()
     {
+        Debug.Log ("Starting coroutine InHouse");
+        aCoroutineRuns = true;
+
         // takes a little while for the player to get inside and disappear
         yield return new WaitForSeconds(0.3f);
-        p.SetActive(false);
+        player.SetActive(false);
         // takes some more time until the player looks outside the window
         yield return new WaitForSeconds(1f);
-        playerRend.sprite = endlevel.playerExposed ? Resources.Load<Sprite>("SmileyPictures/player_exposed") : Resources.Load<Sprite>("SmileyPictures/player_healthy");
         ShowPlayer();
+        playerRend.sprite = endlevel.playerExposed ? Resources.Load<Sprite>("SmileyPictures/player_exposed") : Resources.Load<Sprite>("SmileyPictures/player_healthy");
+
+        //playerRend.sortingLayerName = "Default";
+        isPlayerInside = true;
+
         // a little bit later we notify the end level controller that the player is home
         yield return new WaitForSeconds(0.5f);
+        //FIXME: has to be implemented in endlevel controller base
         endlevel.NotifyPlayerAtHome();
+
+        // maybe the player wants to get out again
+        //if (EnterAndExitAtWill)
+        //{
+        //    // If still running, stop coroutine Set out house to avoid infinite loop
+        //    StopCoroutine (SetPlayerOutOfHouse(p));
+        //    StartCoroutine(SetPlayerOutOfHouse(p));
+        //}
+        Debug.Log("End of Coroutine");
+        aCoroutineRuns = false;
+    }
+
+    void Update ()
+    {
+        // TODO: Only do this every 0.1 secs or so to increase performance
+        if (Time.time - lastTime >= timeBetweenUpdates)
```

Comment:
Time.frameCount will give you the total number of frames since you started the game. so something like how many times the Update method was called since the start. Would it be better to use this instead of a time difference? Since frameCount is counted by unity anyway we would only need something like Time.frameCount%whatsoever. instead of save extra time variables?

Reply:
Thanks for pointing this out, i didnt know that.
In some situations i would still prefer working with time, since it is independent of the frame rate...
Nevertheless, we can remove this whole if since it is only used for debugging output that is no longer needed.